openpype/hosts/houdini/plugins/publish/collect_asset_handles.py

```diff
@@ -0,0 +1,124 @@
+    order = pyblish.api.CollectorOrder + 0.499
+
+    label = "Collect Asset Handles"
+    use_asset_handles = True
+
+    def process(self, instance):
+        # Only process instances without already existing handles data
+        # but that do have frameStartHandle and frameEndHandle defined
+        # like the data collected from CollectRopFrameRange
+        if "frameStartHandle" not in instance.data:
+            return
+        if "frameEndHandle" not in instance.data:
+            return
+
+        has_existing_data = {
+            "handleStart",
+            "handleEnd",
+            "frameStart",
+            "frameEnd"
+        }.issubset(instance.data)
+        if has_existing_data:
+            return
+
+        attr_values = self.get_attr_values_from_data(instance.data)
+        if attr_values.get("use_handles", self.use_asset_handles):
+            asset_data = instance.data["assetEntity"]["data"]
+            handle_start = asset_data.get("handleStart", 0)
+            handle_end = asset_data.get("handleEnd", 0)
+        else:
+            handle_start = 0
+            handle_end = 0
+
+        frame_start = instance.data["frameStartHandle"] + handle_start
+        frame_end = instance.data["frameEndHandle"] - handle_end
+
+        instance.data.update({
+            "handleStart": handle_start,
+            "handleEnd": handle_end,
+            "frameStart": frame_start,
+            "frameEnd": frame_end
+        })
+
+        # Log debug message about the collected frame range
+        if attr_values.get("use_handles", self.use_asset_handles):
+            self.log.debug(
+                "Full Frame range with Handles "
+                "[{frame_start_handle} - {frame_end_handle}]"
+                .format(
+                    frame_start_handle=instance.data["frameStartHandle"],
+                    frame_end_handle=instance.data["frameEndHandle"]
+                )
+            )
+        else:
+            self.log.debug(
+                "Use handles is deactivated for this instance, "
+                "start and end handles are set to 0."
+            )
+
+        # Log collected frame range to the user
+        message = "Frame range [{frame_start} - {frame_end}]".format(
+            frame_start=frame_start,
+            frame_end=frame_end
+        )
+        if handle_start or handle_end:
+            message += " with handles [{handle_start}]-[{handle_end}]".format(
+                handle_start=handle_start,
+                handle_end=handle_end
+            )
+        self.log.info(message)
+
+        if instance.data.get("byFrameStep", 1.0) != 1.0:
+            self.log.info(
+                "Frame steps {}".format(instance.data["byFrameStep"]))
+
+        # Add frame range to label if the instance has a frame range.
+        label = instance.data.get("label", instance.data["name"])
+        instance.data["label"] = (
+            "{label} [{frame_start} - {frame_end}]"
```

Comment:
I wonder if for a Houdini artist it'd be more sensible if the label would show the full ROP range because that's what they are dealing with most transparently. What do you think?

Reply:
Does changing that `debug` into `info` would help ?  
![image](https://github.com/ynput/OpenPype/assets/20871534/e6c8ea9b-274a-4f75-824a-5265ed126cb7)

Reply:
Hmm - it does read clearer to me - just a bit worried we're spamming the user with info if there's like say 10 rop nodes.

Maybe just tweaking the label is sufficient- Maya e.g. also labels with full frame range [here](https://github.com/ynput/OpenPype/blob/59410c7b0c7bae83ce3406bf90e1c9fba583de35/openpype/hosts/maya/plugins/publish/collect_vrayscene.py#L105-L111) and [here](https://github.com/ynput/OpenPype/blob/75e265198bda58d78b1b21c0f9be2713a67b74f2/openpype/hosts/maya/plugins/publish/collect_render.py#L310-L315).

Reply:
Does this communicate well ? I'm going to push my commit shortly.
![image](https://github.com/ynput/OpenPype/assets/20871534/048753a7-f370-4b48-8cec-07113650169f)